main/src/main/scala/sbt/internal/Aggregation.scala

```diff
@@ -114,19 +115,18 @@ object Aggregation {
   )(implicit display: Show[ScopedKey[_]]): State = {
     val complete = timedRun[T](s, ts, extra)
     showRun(complete, show)
-    @tailrec def findReload(
-        incomplete: Incomplete,
-        remaining: List[Incomplete],
-        visited: Set[Incomplete]
-    ): Boolean = {
+    lazy val visited = Collections
```

Comment:
I think it should be noted here in comments that we tried using Set and it caused performance issue due to deeply nested case class doing a fairly straightforward operation.